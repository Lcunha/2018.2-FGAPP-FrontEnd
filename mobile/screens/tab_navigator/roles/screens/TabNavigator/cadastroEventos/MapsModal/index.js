import React, { Component } from "react";
import {
	Card,
	CardItem,
	Body,
	Icon,
	Button,
	Input,
	Form,
	Item
} from "native-base";
import {
	Text,
	StyleSheet,
	View,
	ScrollView,
	Alert,
	KeyboardAvoidingView,
	ToastAndroid
} from "react-native";
import Modal from "react-native-modal";
import { MapView, Location, Permissions } from "expo";
import { Marker } from "react-native-maps";

const latLngDelta = 0.003;

export default class MapsModal extends Component {
	state = {
		searchKeyWord: "",
		address: {},
		coordenadaRole: {
			//Valores Padrões para Latitude e Longitude. Servem para caso a informação dada de localização não seja válida
			latitude: -15.7856594,
			longitude: -47.8959588,
			latitudeDelta: latLngDelta,
			longitudeDelta: latLngDelta
		},
		region: {
			latitude: 0,
			longitude: 0,
			latitudeDelta: latLngDelta,
			longitudeDelta: latLngDelta
		},
		initialRegion: {
			latitude: -15.989373048115894,
			longitude: -48.044495824724436,
			latitudeDelta: latLngDelta,
			longitudeDelta: latLngDelta
		}
	};

	componentDidMount() {
		Permissions.askAsync(Permissions.LOCATION);
	}

	_getLocalizationFromSearch = async key => {
		try {
			let result = await Location.geocodeAsync(key);
			if (result.length > 0) {
				this._mapView.animateToCoordinate({
					latitude: result[0].latitude,
					longitude: result[0].longitude,
					latitudeDelta: latLngDelta,
					longitudeDelta: latLngDelta
				});
			} else {
				Alert.alert(
					"Endereço Inválido",
					"Não foi encontrado uma localização com este endereço."
				);
			}
		} catch (e) {
			this.setState({ error: e });
		}
	};

	_getAddressFromCoordinates = async coordinates => {
		try {
			let result = await Location.reverseGeocodeAsync({
				latitude: coordinates.latitude,
				longitude: coordinates.longitude
			});
			this.setState({ address: result[0] });
		} catch (e) {
			this.setState({ error: e });
		}
	};

	_onRegionChange(region) {
		this.setState({ region });
		// console.log("REGION:", region);
	}

	_handleBtnOk() {
		this.props.onLocationSelection(this.state.region);
		this.setState({ initialRegion: this.state.region });
	}

	_handleEscape() {
		ToastAndroid.show(
			"Você deve selecionar um local e dar OK!",
			ToastAndroid.SHORT
		);
	}

	render() {
		return (
			<Modal
				isVisible={this.props.visible}
				onBackButtonPress={() => this._handleEscape()}
				onBackdropPress={() => this._handleEscape()}
				animationIn="fadeIn"
				animationOut="fadeOut"
			>
				<Card style={styles.modalCardInput}>
					<ScrollView
						keyboardShouldPersistTaps="always"
						keyboardDismissMode="on-drag"
					>
						<View>
							<Text style={{ margin: 10, textAlign: "center" }}>
								Informe o nome do local do evento para
								pesquisar:
							</Text>

							<View
								width="100%"
								flexDirection="row"
								borderWidth={1}
								borderColor="lightgrey"
							>
								<Input
									placeholder="Pesquisar Endereço"
									multiline={false}
									width="90%"
									style={{ textAlign: "center" }}
									onChangeText={text =>
										this.setState({
											searchKeyWord: text
										})
									}
								/>
								<Button
									transparent
									onPress={() => {
										this._getLocalizationFromSearch(
											this.state.searchKeyWord
										);
									}}
								>
									<Icon
										name="search"
										style={{ color: "black" }}
									/>
								</Button>
							</View>

							<MapView
								ref={ref => (this._mapView = ref)}
								style={{ height: 200, width: "100%" }}
								initialRegion={this.state.initialRegion}
								onRegionChangeComplete={region => {
									this._onRegionChange(region);
									this._getAddressFromCoordinates(region);
								}}
							>
								<Marker
									coordinate={{
										latitude: this.state.region.latitude,
										longitude: this.state.region.longitude
									}}
									title={this.props.placeName}
									description={"Local do Rolê"}
								/>
							</MapView>
							<Text
								style={{
									alignSelf: "center",
									fontSize: 10,
									color: "grey",
									marginBottom: 10
								}}
							>
								{`${this.state.region.latitude} \ ${
									this.state.region.longitude
								}`}
							</Text>

							<Text
								style={{ textAlign: "center", color: "grey" }}
							>
								{this.state.address.city != null &&
									this.state.address.city + ", "}
								{this.state.address.street != null &&
									this.state.address.street !=
										"Unnamed Road" &&
									this.state.address.street + ", "}
								{this.state.address.name != null &&
									this.state.address.name != "Unnamed Road" &&
									this.state.address.name + "\n"}
								{this.state.address.region != null &&
									this.state.address.region + " - "}
								{this.state.address.country != null &&
									this.state.address.country + "\n"}
								{this.state.address.postalCode != null &&
									this.state.address.postalCode + "\n"}
							</Text>
						</View>

						<View
							width="100%"
							margin={5}
							borderWidth={1}
							borderColor="rgba(0,0,0,0.1)"
							alignSelf="center"
						/>

						<View marginBottom={10}>
							<Text style={{ textAlign: "center", margin: 10 }}>
								O local apresentado no mapa está correto?
							</Text>

							<Button
								block
								success
								style={styles.btn}
								onPress={() => this._handleBtnOk()}
							>
								<Text style={{ color: "white" }}>
									Sim, é ali mesmo!
								</Text>
							</Button>
						</View>
					</ScrollView>
				</Card>
			</Modal>
		);
	}
}

const styles = StyleSheet.create({
	modalCardInput: {
		backgroundColor: "white",
		borderRadius: 10,
		width: "100%",
		height: "80%",
		alignSelf: "stretch",
		alignItems: "stretch"
	},
	btn: {
		width: "90%",
		alignSelf: "center",
		borderRadius: 10
	}
});
