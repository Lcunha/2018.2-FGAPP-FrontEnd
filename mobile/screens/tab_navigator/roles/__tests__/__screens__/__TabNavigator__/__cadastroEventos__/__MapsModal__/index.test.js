import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import MapsModal from '../../../../../screens/TabNavigator/cadastroEventos/MapsModal/index';

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer.create(<MapsModal/>).toJSON();
  expect(tree).toMatchSnapshot();
});