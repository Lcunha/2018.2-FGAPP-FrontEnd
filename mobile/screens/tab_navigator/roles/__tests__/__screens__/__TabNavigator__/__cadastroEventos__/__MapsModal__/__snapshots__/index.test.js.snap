// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onRequestClose={[Function]}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
    testID={undefined}
  />
  <View
    collapsable={undefined}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    pointerEvents="box-none"
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
            "borderLeftWidth": 0.5,
            "borderRadius": 2,
            "borderRightWidth": 0.5,
            "borderTopWidth": 0.5,
            "elevation": 3,
            "flexWrap": "nowrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          },
          Object {
            "alignItems": "stretch",
            "alignSelf": "stretch",
            "backgroundColor": "white",
            "borderRadius": 10,
            "height": "80%",
            "width": "100%",
          },
        ]
      }
      virtual={undefined}
    >
      <RCTScrollView
        keyboardDismissMode="on-drag"
        keyboardShouldPersistTaps="always"
      >
        <View>
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "margin": 10,
                  "textAlign": "center",
                }
              }
            >
              Informe o nome do local do evento para pesquisar:
            </Text>
            <View
              borderColor="lightgrey"
              borderWidth={1}
              flexDirection="row"
              width="100%"
            >
              <TextInput
                allowFontScaling={true}
                editable={true}
                multiline={false}
                onChangeText={[Function]}
                placeholder="Pesquisar Endereço"
                placeholderTextColor="#575757"
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "flex": 1,
                      "fontSize": 17,
                      "height": 50,
                      "paddingLeft": 5,
                      "paddingRight": 5,
                    },
                    Object {
                      "textAlign": "center",
                    },
                  ]
                }
                underlineColorAndroid="rgba(0,0,0,0)"
                virtual={undefined}
                width="90%"
              />
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "flex-start",
                    "backgroundColor": "transparent",
                    "borderBottomWidth": null,
                    "borderColor": "#007aff",
                    "borderLeftWidth": null,
                    "borderRadius": 5,
                    "borderRightWidth": null,
                    "borderTopWidth": null,
                    "borderWidth": undefined,
                    "elevation": 0,
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "space-between",
                    "opacity": 1,
                    "paddingBottom": 6,
                    "paddingTop": 6,
                    "shadowColor": null,
                    "shadowOffset": null,
                    "shadowOpacity": null,
                    "shadowRadius": null,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
              </View>
            </View>
            <MapView
              initialRegion={
                Object {
                  "latitude": -15.989373048115894,
                  "latitudeDelta": 0.003,
                  "longitude": -48.044495824724436,
                  "longitudeDelta": 0.003,
                }
              }
              onRegionChangeComplete={[Function]}
              style={
                Object {
                  "height": 200,
                  "width": "100%",
                }
              }
            >
              <Marker
                coordinate={
                  Object {
                    "latitude": 0,
                    "longitude": 0,
                  }
                }
                description="Local do Rolê"
                title={undefined}
              />
            </MapView>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "color": "grey",
                  "fontSize": 10,
                  "marginBottom": 10,
                }
              }
            >
              0  0
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "grey",
                  "textAlign": "center",
                }
              }
            />
          </View>
          <View
            alignSelf="center"
            borderColor="rgba(0,0,0,0.1)"
            borderWidth={1}
            margin={5}
            width="100%"
          />
          <View
            marginBottom={10}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "margin": 10,
                  "textAlign": "center",
                }
              }
            >
              O local apresentado no mapa está correto?
            </Text>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#5cb85c",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 10,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "center",
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "shadowColor": undefined,
                  "shadowOffset": undefined,
                  "shadowOpacity": undefined,
                  "shadowRadius": undefined,
                  "width": "90%",
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Sim, é ali mesmo!
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</Modal>
`;
